### ISSUE TITLE
Add a representation of JID objects

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3156]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#3156|
|Type|enhancement|
|Created|2008-04-03 12:54:13Z|
|Branch|https://github.com/twisted/twisted/tree/jid-repr-3156|

`twisted.words.protocols.jabber.jid.JID` objects currently lack a useful representation, defaulting to the class name and its internal python ID. The attached patch adds a `__repr__` method to the JID class that returns a string matching how you could recreate the same JID object.

Attachments:

* [jid_repr.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/45d/45d054cba06f055820ca4ab1b164d78bcdf2115c/a377f30e01da4b3cedf7c617136c2655ee5df187.patch) (359 bytes) - added by ralphm on 2008-04-03 12:56:00Z - Patch adding a __repr__ method to the JID class

<details><summary>Searchable metadata</summary>

```
trac-id__3156 3156
type__enhancement enhancement
reporter__ralphm ralphm
priority__highest highest
milestone__ 
branch__branches_jid_repr_3156 branches/jid-repr-3156
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1207227253000000 1207227253000000
changetime__1207315680000000 1207315680000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to @ralphm|
|-|-|

Please add test coverage

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm commented|
|-|-|

(In [23187]) Branching to 'jid-repr-3156'

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm set owner to @exarkun|
|-|-|

*ahem*

See the branch, including a test and docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @ralphm|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set status to `closed`|
|-|-|

(In [23197]) Add a formal string representation to JID objects using __repr__.

Author: ralphm
Reviewer: therve
Fixes [#3156](https://github.com/twisted/twisted/issues/3156).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
